services/src/datasets/add_from_directory.rs

```diff
@@ -36,9 +37,12 @@ pub async fn add_datasets_from_directory<S: MockableSession, D: DatasetDb<S>>(
         Ok(())
     }
 
-    let dir = fs::read_dir(file_path);
+    let dir = fs::read_dir(&file_path);
     if dir.is_err() {
```

Comment:
```Rust
let Ok(dir) = else {
    … output to logs …
    return;
};
```

and then…

Reply:
I applied that new pattern 8ccc34bde0a3e09e5cd4346a9044f2bc7fbd9535